packages/dx-react-scheduler/src/plugins/all-day-panel.tsx

```diff
@@ -33,44 +30,55 @@ const CellPlaceholder = params => <TemplatePlaceholder name="allDayPanelCell" pa
 
 class AllDayPanelBase extends React.PureComponent<AllDayPanelProps, AllDayPanelState> {
   state: AllDayPanelState = {
-    tableRef: null,
+    rects: [],
+    elementsMeta: {},
   };
   static defaultProps: Partial<AllDayPanelProps> = {
     messages: {},
   };
   static components: PluginComponents = {
     appointmentLayerComponent: 'AppointmentLayer',
     layoutComponent: 'Layout',
+    layoutContainerComponent: 'LayoutContainer',
     cellComponent: 'Cell',
     rowComponent: 'Row',
     titleCellComponent: 'TitleCell',
     containerComponent: 'Container',
   };
 
-  allDayPanelRef = (ref) => {
-    this.setState({
-      tableRef: ref,
-    });
-  }
+  allDayCellsData = memoize(viewCellsData => allDayCells(viewCellsData));
```

Comment:
This function should be extracted outside the class since it doesn't use its members